File: proto1/CUSalonCollection.m
////
////  CUSalonCollection.m
////  proto1
////
////  Created by Connor Reid on 1/29/15.
////  Copyright (c) 2015 test. All rights reserved.
////
//
//#import "CUSalonCollection.h"
//#import "CUUtilities.h"
//#import "CUSalon.h"
//#import "CUDesigner.h"
//
//@implementation CUSalonCollection
//
//static NSMutableArray *_salons;
//
//+ (NSMutableArray*)getRandomDefaultTimeSlots
//{
//    NSMutableArray *randomDefaultTimeSlots = [NSMutableArray array];
//    
//    for (NSInteger i = 0; i < 48; ++i)
//    {
//        BOOL isAvailable = NO;
//        int tmp = (arc4random() % 30) +1;
//        if (tmp % 5 == 0)
//        {
//            isAvailable = YES;
//        }
//        [randomDefaultTimeSlots addObject:[NSNumber numberWithBool:isAvailable]];
//    }
//    
//    return randomDefaultTimeSlots;
//}
//
//+ (NSDictionary*)testSalonDictionaries
//{
//    id review = [[NSObject alloc] init];
//    NSArray *reviews = @[review];
//    
//    id reservation = [[NSObject alloc] init];
//    NSArray *reservations = @[reservation];
//    
//    NSDictionary *testDesignerDictionary = @{@"guid"                        :[CUUtilities createGUID],
//                                             @"salonGuid"                   :[CUUtilities createGUID],
//                                             @"name:"                       :@"Bobby Weave",
//                                             @"email"                       :@"bobbyweave@test.com",
//                                             @"defaultAvailableTimeSlots"   :@{@"monday"    :[CUSalonCollection getRandomDefaultTimeSlots],
//                                                                               @"tuesday"   :[CUSalonCollection getRandomDefaultTimeSlots],
//                                                                               @"wednesday" :[CUSalonCollection getRandomDefaultTimeSlots],
//                                                                               @"thursday"  :[CUSalonCollection getRandomDefaultTimeSlots],
//                                                                               @"friday"    :[CUSalonCollection getRandomDefaultTimeSlots],
//                                                                               @"saturday"  :[CUSalonCollection getRandomDefaultTimeSlots],
//                                                                               @"sunday"    :[CUSalonCollection getRandomDefaultTimeSlots]},
//                                             @"customAvailableTimeSlots"    :@{@"1451635200":@YES}};
//    
//    NSDictionary *testDesignersDictionary01 = @{@"designers":@{testDesignerDictionary[@"guid"]:testDesignerDictionary}};
//
//    NSDictionary *testSalonDictionary01 = @{@"images":@[[UIImage imageNamed:@"salon1.jpeg"]],
//                                             @"hours":@{@"monday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"tuesday"      :@[@"9:00AM", @"6:00PM"],
//                                                        @"wednesday"    :@[@"9:00AM", @"6:00PM"],
//                                                        @"thursday"     :@[@"9:00AM", @"6:00PM"],
//                                                        @"friday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"saturday"     :@[@"10:00AM", @"5:00PM"],
//                                                        @"sunday"       :@[@"12:00PM", @"4:00PM"]},
//                                              @"name":@"박철 헤어",
//                                          @"location":@{@"latitude":@30.513925, @"longitude":@97.688555},    //  30.513925, -97.688555 -> test location (location of old town square, rr, tx)
//                                           @"address":@"",
//                                           @"reviews":reviews,
//                                            @"rating":[NSNumber numberWithFloat:4.5],
//                                       @"phoneNumber":@"512-555-5555",
//                                              @"guid":[CUUtilities createGUID],
//                                      @"reservations":reservations,
//                                         @"designers":testDesignersDictionary01};
//
//    NSDictionary *testSalonDictionary02 = @{@"images":@[[UIImage imageNamed:@"salon2.jpeg"]],
//                                             @"hours":@{@"monday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"tuesday"      :@[@"9:00AM", @"6:00PM"],
//                                                        @"wednesday"    :@[@"9:00AM", @"6:00PM"],
//                                                        @"thursday"     :@[@"9:00AM", @"6:00PM"],
//                                                        @"friday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"saturday"     :@[@"10:00AM", @"5:00PM"],
//                                                        @"sunday"       :@[@"12:00PM", @"4:00PM"]},
//                                              @"name":@"CNN 헤어",
//                                          @"location":@{@"latitude":@30.513925, @"longitude":@97.688555},    //  30.513925, -97.688555 -> test location (location of old town square, rr, tx)
//                                           @"address":@"",
//                                           @"reviews":reviews,
//                                            @"rating":[NSNumber numberWithFloat:4.5],
//                                       @"phoneNumber":@"512-555-5555",
//                                              @"guid":[CUUtilities createGUID],
//                                      @"reservations":reservations,
//                                         @"designers":testDesignersDictionary01};
//    
//    NSDictionary *testSalonDictionary03 = @{@"images":@[[UIImage imageNamed:@"salon3.jpeg"]],
//                                             @"hours":@{@"monday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"tuesday"      :@[@"9:00AM", @"6:00PM"],
//                                                        @"wednesday"    :@[@"9:00AM", @"6:00PM"],
//                                                        @"thursday"     :@[@"9:00AM", @"6:00PM"],
//                                                        @"friday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"saturday"     :@[@"10:00AM", @"5:00PM"],
//                                                        @"sunday"       :@[@"12:00PM", @"4:00PM"]},
//                                              @"name":@"박승철 헤어",
//                                          @"location":@{@"latitude":@30.513925, @"longitude":@97.688555},    //  30.513925, -97.688555 -> test location (location of old town square, rr, tx)
//                                           @"address":@"",
//                                           @"reviews":reviews,
//                                            @"rating":[NSNumber numberWithFloat:4.5],
//                                       @"phoneNumber":@"512-555-5555",
//                                              @"guid":[CUUtilities createGUID],
//                                      @"reservations":reservations,
//                                         @"designers":testDesignersDictionary01};
//    
//    NSDictionary *testSalonDictionary04 = @{@"images":@[[UIImage imageNamed:@"salon4.jpeg"]],
//                                             @"hours":@{@"monday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"tuesday"      :@[@"9:00AM", @"6:00PM"],
//                                                        @"wednesday"    :@[@"9:00AM", @"6:00PM"],
//                                                        @"thursday"     :@[@"9:00AM", @"6:00PM"],
//                                                        @"friday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"saturday"     :@[@"10:00AM", @"5:00PM"],
//                                                        @"sunday"       :@[@"12:00PM", @"4:00PM"]},
//                                              @"name":@"UFO 헤어",
//                                          @"location":@{@"latitude":@30.513925, @"longitude":@97.688555},    //  30.513925, -97.688555 -> test location (location of old town square, rr, tx)
//                                           @"address":@"",
//                                           @"reviews":reviews,
//                                            @"rating":[NSNumber numberWithFloat:4.5],
//                                       @"phoneNumber":@"512-555-5555",
//                                              @"guid":[CUUtilities createGUID],
//                                      @"reservations":reservations,
//                                         @"designers":testDesignersDictionary01};
//    
//    NSDictionary *testSalonDictionary05 = @{@"images":@[[UIImage imageNamed:@"salon5.jpeg"]],
//                                             @"hours":@{@"monday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"tuesday"      :@[@"9:00AM", @"6:00PM"],
//                                                        @"wednesday"    :@[@"9:00AM", @"6:00PM"],
//                                                        @"thursday"     :@[@"9:00AM", @"6:00PM"],
//                                                        @"friday"       :@[@"9:00AM", @"6:00PM"],
//                                                        @"saturday"     :@[@"10:00AM", @"5:00PM"],
//                                                        @"sunday"       :@[@"12:00PM", @"4:00PM"]},
//                                              @"name":@"FarWest Hair",
//                                          @"location":@{@"latitude":@30.513925, @"longitude":@97.688555},    //  30.513925, -97.688555 -> test location (location of old town square, rr, tx)
//                                           @"address":@"",
//                                           @"reviews":reviews,
//                                            @"rating":[NSNumber numberWithFloat:4.5],
//                                       @"phoneNumber":@"512-555-5555",
//                                              @"guid":[CUUtilities createGUID],
//                                      @"reservations":reservations,
//                                         @"designers":testDesignersDictionary01};
//    
//    NSDictionary *testSalonsDictionary = @{@"salons":@{testSalonDictionary01[@"guid"]:testSalonDictionary01,
//                                                       testSalonDictionary02[@"guid"]:testSalonDictionary02,
//                                                       testSalonDictionary03[@"guid"]:testSalonDictionary03,
//                                                       testSalonDictionary04[@"guid"]:testSalonDictionary04,
//                                                       testSalonDictionary05[@"guid"]:testSalonDictionary05,}};
//    
//    return testSalonsDictionary;
//}
//
//+ (void)set_g_Salons:(NSDictionary*)salonsDictionary
//{
//    static dispatch_once_t onceToken;
//    dispatch_once(&onceToken, ^
//    {
//        _salons = [[NSMutableArray alloc] init];
//    });
//    [_salons removeAllObjects];
//    NSDictionary *salons = salonsDictionary[@"salons"];
//    for (id key in salons)
//    {
//        CUSalon *salon = [CUSalon dictionaryToSalon:salons[key]];
//        [_salons addObject:salon];
//    }
//}
//
//+ (id)g_Salons
//{
//    if (_salons)
//    {
//        return _salons;
//    }
//    else
//    {
//        return NULL;
//    }
//}
//
//@end

File: proto1/CUSalonCollection.swift
//
//  CUSalonCollectio.swift
//  CurlUpMobileIOS
//
//  Created by Ian on 2/7/15.
//  Copyright (c) 2015 test. All rights reserved.
//

//
//  CUSalonCollection.swift
//  CurlUpMobileIOS
//
//  Created by Connor Reid on 2/2/15.
//  Copyright (c) 2015 test. All rights reserved.
//

import Foundation
import UIKit
import CoreLocation

@objc class CUSalonCollection: NSObject {
    var salons: [CUSalon]
    override init() {
        self.salons = [CUSalon]()
        super.init()
    }
    
    class var sharedInstance: CUSalonCollection {
        struct Singleton
        {
            static let _singleton: CUSalonCollection = CUSalonCollection()
        }
        return Singleton._singleton
    }
    
    // For Objc
    func set_g_Salons(fromDictionary objCDictionary: NSDictionary) {
        //        let decoder: NSCoder = NSCoder.alloc()
        //        let salons = decoder.decodeObjectOfClass(NSDictionary.self, forKey: "salons") as Dictionary<String, NSDictionary>
        let salonsDictionary: [String: NSDictionary] = objCDictionary as [String: NSDictionary]
        let salons: [String: NSDictionary] = salonsDictionary["salons"] as [String: NSDictionary]
        println(salons)
    }
    
    func g_Salons() -> NSDictionary {
        var dictionary: NSDictionary = NSDictionary.alloc()
        
        return dictionary
    }
    
    func objcSalonsDictionaryToSwiftSalonsDictionary(fromSalonsDictionary objcSalonsDictionary: Dictionary<String, NSObject>) -> Dictionary<String, Array<CUSalon>> {
        var salonsDictionary: [String: [CUSalon]] = [String: [CUSalon]]()
        return salonsDictionary
    }
    
    func _set_g_Salons(fromDictionary salonsDictionary: [CUSalon]) {
        CUSalonCollection.sharedInstance.salons = [CUSalon]()
        if let _salons: [CUSalon] = salonsDictionary as [CUSalon]? {
            for salon in _salons {
                CUSalonCollection.sharedInstance.salons.append(salon)
            }
        }
    }
    
    func getRandomDefaultTimeSlots() -> [Bool] {
        var defaultTimeSlots: [Bool] = [Bool]();
        
        var isAvailable = false;
        var tmp = (arc4random() % 30) + 1;
        if (tmp % 5 == 0)
        {
            isAvailable = true;
        }
        defaultTimeSlots.append(isAvailable);
        
        return defaultTimeSlots;
    }
    
    func getTestSalonsDictionary() -> [CUSalon] {
        let salons = [CUSalon(images        :[UIImage(named:"salon1.jpeg")!],
            hours         :["monday"    :["9:00AM", "6:00PM"],
                "tuesday"   :["9:00AM", "6:00PM"],
                "wednesday" :["9:00AM", "6:00PM"],
                "thursday"  :["9:00AM", "6:00PM"],
                "friday"    :["9:00AM", "6:00PM"],
                "saturday"  :["10:00AM", "5:00PM"],
                "sunday"    :["12:00PM", "4:00PM"]],
            name          :"박철 헤어",
            location      :CLLocation(latitude:30.513925,
                longitude:97.688555)!,
            address       :"111 Somewhere, USA",
            reviews       :[0],
            rating        :4.5,
            phoneNumber   :"512-555-5555",
            guid          :NSUUID().UUIDString,
            reservations  :[CUReservation(guid              :NSUUID().UUIDString,
                date              :NSDate(),
                pictures          :[UIImage(named: "salon1.jpeg")!],
                haircutDescription:"Two-Block",
                clientGuid        :NSUUID().UUIDString,
                clientEmail       :"client01@test.com",
                clientName        :"Client One",
                clientPhoneNumber :"555-555-5555",
                salonGuid         :NSUUID().UUIDString,
                designerGuid      :NSUUID().UUIDString)],
            designers     :[CUDesigner(guid                      :NSUUID().UUIDString,
                salonGuid                 :NSUUID().UUIDString,
                name                      :"Bobby Weave",
                email                     :"bobby@weave.test",
                phoneNumber               :"555-555-5555",
                defaultAvailableTimeSlots :["monday"     :getRandomDefaultTimeSlots() as [Bool],
                    "tuesday"    :getRandomDefaultTimeSlots() as [Bool],
                    "wednesday"  :getRandomDefaultTimeSlots() as [Bool],
                    "thursday"   :getRandomDefaultTimeSlots() as [Bool],
                    "friday"     :getRandomDefaultTimeSlots() as [Bool],
                    "saturday"   :getRandomDefaultTimeSlots() as [Bool],
                    "sunday"     :getRandomDefaultTimeSlots() as [Bool]],
                customAvaiableTimeSlots   :[1451635200:true])]),
            CUSalon(images        :[UIImage(named:"salon2.jpeg")!],
                hours         :["monday"      :["9:00AM", "6:00PM"],
                    "tuesday"     :["9:00AM", "6:00PM"],
                    "wednesday"   :["9:00AM", "6:00PM"],
                    "thursday"    :["9:00AM", "6:00PM"],
                    "friday"      :["9:00AM", "6:00PM"],
                    "saturday"    :["10:00AM", "5:00PM"],
                    "sunday"      :["12:00PM", "4:00PM"]],
                name          :"CNN 헤어",
                location      :CLLocation(latitude:30.513925,
                    longitude:97.688555)!,
                address       :"777 Seoul, Korea",
                reviews       :[0],
                rating        :4.5,
                phoneNumber   :"512-555-5555",
                guid          :NSUUID().UUIDString,
                reservations  :[CUReservation(guid              :NSUUID().UUIDString,
                    date              :NSDate(),
                    pictures          :[UIImage(named: "salon1.jpeg")!],
                    haircutDescription:"Two-Block",
                    clientGuid        :NSUUID().UUIDString,
                    clientEmail       :"client01@test.com",
                    clientName        :"Client One",
                    clientPhoneNumber :"555-555-5555",
                    salonGuid         :NSUUID().UUIDString,
                    designerGuid      :NSUUID().UUIDString)],
                designers     :[CUDesigner(guid                      :NSUUID().UUIDString,
                    salonGuid                 :NSUUID().UUIDString,
                    name                      :"Bobby Weave",
                    email                     :"bobby@weave.test",
                    phoneNumber               :"555-555-5555",
                    defaultAvailableTimeSlots :["monday"     :getRandomDefaultTimeSlots() as [Bool],
                        "tuesday"    :getRandomDefaultTimeSlots() as [Bool],
                        "wednesday"  :getRandomDefaultTimeSlots() as [Bool],
                        "thursday"  :getRandomDefaultTimeSlots() as [Bool],
                        "friday"    :getRandomDefaultTimeSlots() as [Bool],
                        "saturday"  :getRandomDefaultTimeSlots() as [Bool],
                        "sunday"    :getRandomDefaultTimeSlots() as [Bool]],
                    customAvaiableTimeSlots   :[1451635200:true])]),
            CUSalon(images        :[UIImage(named:"salon3.jpeg")!],
                hours         :["monday"    :["9:00AM", "6:00PM"],
                    "tuesday"   :["9:00AM", "6:00PM"],
                    "wednesday" :["9:00AM", "6:00PM"],
                    "thursday"  :["9:00AM", "6:00PM"],
                    "friday"    :["9:00AM", "6:00PM"],
                    "saturday"  :["10:00AM", "5:00PM"],
                    "sunday"    :["12:00PM", "4:00PM"]],
                name          :"박철 헤어",
                location      :CLLocation(latitude:30.513925,
                    longitude:97.688555)!,
                address       :"8844 Two-block Blvd",
                reviews       :[0],
                rating        :4.5,
                phoneNumber   :"512-555-5555",
                guid          :NSUUID().UUIDString,
                reservations  :[CUReservation(guid              :NSUUID().UUIDString,
                    date              :NSDate(),
                    pictures          :[UIImage(named: "salon1.jpeg")!],
                    haircutDescription:"Two-Block",
                    clientGuid        :NSUUID().UUIDString,
                    clientEmail       :"client01@test.com",
                    clientName        :"Client One",
                    clientPhoneNumber :"555-555-5555",
                    salonGuid         :NSUUID().UUIDString,
                    designerGuid      :NSUUID().UUIDString)],
                designers     :[CUDesigner(guid                      :NSUUID().UUIDString,
                    salonGuid                 :NSUUID().UUIDString,
                    name                      :"Bobby Weave",
                    email                     :"bobby@weave.test",
                    phoneNumber               :"555-555-5555",
                    defaultAvailableTimeSlots :["monday"    :getRandomDefaultTimeSlots() as [Bool],
                        "tuesday"   :getRandomDefaultTimeSlots() as [Bool],
                        "wednesday" :getRandomDefaultTimeSlots() as [Bool],
                        "thursday"  :getRandomDefaultTimeSlots() as [Bool],
                        "friday"    :getRandomDefaultTimeSlots() as [Bool],
                        "saturday"  :getRandomDefaultTimeSlots() as [Bool],
                        "sunday"    :getRandomDefaultTimeSlots() as [Bool]],
                    customAvaiableTimeSlots   :[1451635200:true])]),
            CUSalon(images        :[UIImage(named:"salon4.jpeg")!],
                hours         :["monday"    :["9:00AM", "6:00PM"],
                    "tuesday"   :["9:00AM", "6:00PM"],
                    "wednesday" :["9:00AM", "6:00PM"],
                    "thursday"  :["9:00AM", "6:00PM"],
                    "friday"    :["9:00AM", "6:00PM"],
                    "saturday"  :["10:00AM", "5:00PM"],
                    "sunday"    :["12:00PM", "4:00PM"]],
                name          :"박철 헤어",
                location      :CLLocation(latitude:30.513925,
                    longitude:97.688555)!,
                address       :"555 Five St.",
                reviews       :[0],
                rating        :4.5,
                phoneNumber   :"512-555-5555",
                guid          :NSUUID().UUIDString,
                reservations  :[CUReservation(guid              :NSUUID().UUIDString,
                    date              :NSDate(),
                    pictures          :[UIImage(named: "salon1.jpeg")!],
                    haircutDescription:"Two-Block",
                    clientGuid        :NSUUID().UUIDString,
                    clientEmail       :"client01@test.com",
                    clientName        :"Client One",
                    clientPhoneNumber :"555-555-5555",
                    salonGuid         :NSUUID().UUIDString,
                    designerGuid      :NSUUID().UUIDString)],
                designers     :[CUDesigner(guid                      :NSUUID().UUIDString,
                    salonGuid                 :NSUUID().UUIDString,
                    name                      :"Bobby Weave",
                    email                     :"bobby@weave.test",
                    phoneNumber               :"555-555-5555",
                    defaultAvailableTimeSlots :["monday"     :getRandomDefaultTimeSlots() as [Bool],
                        "tuesday"    :getRandomDefaultTimeSlots() as [Bool],
                        "wednesday"  :getRandomDefaultTimeSlots() as [Bool],
                        "thursday"   :getRandomDefaultTimeSlots() as [Bool],
                        "friday"     :getRandomDefaultTimeSlots() as [Bool],
                        "saturday"   :getRandomDefaultTimeSlots() as [Bool],
                        "sunday"     :getRandomDefaultTimeSlots() as [Bool]],
                    customAvaiableTimeSlots   :[1451635200:true])]),
            CUSalon(images        :[UIImage(named:"salon5.jpeg")!],
                hours         :["monday"      :["9:00AM", "6:00PM"],
                    "tuesday"     :["9:00AM", "6:00PM"],
                    "wednesday"   :["9:00AM", "6:00PM"],
                    "thursday"    :["9:00AM", "6:00PM"],
                    "friday"      :["9:00AM", "6:00PM"],
                    "saturday"    :["10:00AM", "5:00PM"],
                    "sunday"      :["12:00PM", "4:00PM"]],
                name          :"박철 헤어",
                location      :CLLocation(latitude:30.513925,
                    longitude:97.688555)!,
                address       :"333 Triple Rd.",
                reviews       :[0],
                rating        :4.5,
                phoneNumber   :"512-555-5555",
                guid          :NSUUID().UUIDString,
                reservations  :[CUReservation(guid              :NSUUID().UUIDString,
                    date              :NSDate(),
                    pictures          :[UIImage(named: "salon1.jpeg")!],
                    haircutDescription:"Two-Block",
                    clientGuid        :NSUUID().UUIDString,
                    clientEmail       :"client01@test.com",
                    clientName        :"Client One",
                    clientPhoneNumber :"555-555-5555",
                    salonGuid         :NSUUID().UUIDString,
                    designerGuid      :NSUUID().UUIDString)],
                designers     :[CUDesigner(guid                      :NSUUID().UUIDString,
                    salonGuid                 :NSUUID().UUIDString,
                    name                      :"Bobby Weave",
                    email                     :"bobby@weave.test",
                    phoneNumber               :"555-555-5555",
                    defaultAvailableTimeSlots :["monday"     :getRandomDefaultTimeSlots() as [Bool],
                        "tuesday"    :getRandomDefaultTimeSlots() as [Bool],
                        "wednesday"  :getRandomDefaultTimeSlots() as [Bool],
                        "thursday"   :getRandomDefaultTimeSlots() as [Bool],
                        "friday"     :getRandomDefaultTimeSlots() as [Bool],
                        "saturday"   :getRandomDefaultTimeSlots() as [Bool],
                        "sunday"     :getRandomDefaultTimeSlots() as [Bool]],
                    customAvaiableTimeSlots   :[1451635200:true])])]
        return salons
    }

}

